### ISSUE TITLE
[Clean-up] Github App Tech

### ISSUE THREAD

**Issue by KDwevedi** [Code4GovTech/server#14]
[Handover Doc**](https://docs.google.com/document/d/1BN_KdL5tGgNP9-t27jMEum4BSbbxIhT3y188gMihk6s/edit#heading=h.hxaoonavs3p0)

# Description
An anchor ticket to describe the current state and vision for C4GT Github App

# What is this app supposed to be?
The C4GT is the backend for the [C4GT Community Support App](https://github.com/apps/c4gt-community-support/).
It is primarily an event driven application based around [Github Webhooks](https://docs.github.com/en/webhooks/webhook-events-and-payloads) received via a [Github App Installation](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/using-webhooks-with-github-apps)

# TODO
## Documentation
- [x] Proposed/current architecture
- [x] Module level functional documentation
- [ ] Module level vision documentation
- [x] Tech used

## Code Refactor
- [ ] [`/github/events`](https://github.com/Code4GovTech/server/blob/ed561794f76756687afd21d34c71221c5c8e67b0/app.py#L144)
  - [ ] Event Ingestor/Consumer segregation
  - [ ] Data Models and Validators
  - [ ] Singleton Helper class for DB interface 
  - [ ] Extrapolate configuration into classes/json/yaml
    - [ ] App config
    - [ ] comment creator function??
- [ ] Switch to GraphQL 
- [ ] Add observability
- [ ] Tests
## Security
- [ ] [Add Auth](https://docs.github.com/en/webhooks/using-webhooks/best-practices-for-using-webhooks)
- [ ] [Setup RLS for database](https://supabase.com/docs/guides/auth/row-level-security)
- [ ] Should we consider [self-hosting](https://supabase.com/docs/guides/self-hosting)?
## Maintainability
- [ ] Deploy via Jenkins and github actions
- [ ] Dev environment
- [ ] [gh token](https://github.com/Code4GovTech/discord-bot/issues/7)
### References
[Blueprints](https://pgjones.gitlab.io/quart/how_to_guides/blueprints.html)
[Blueprints](https://realpython.com/flask-blueprint/#what-a-flask-application-looks-like)

**Comment by KDwevedi** [comment]
## Functional Requirements for the Github Support App

### General Requirements
- Basic Observability (/health + relevant logs)
- Auth

### Contributor Registration
- Contributor Registration Flow

### C4GT Ticket Handling
- Get Issues
  - When New Repo is Added
- Markdown Parser for Issue Contents
- Feedback on Listed Tickets through Github Comments

### Data Updation
- Record Data of Interest
  - Current Installations
  - Existing Comments
  -  

### VC
- Verifying Snippets
- Updating Snippets

**Comment by KDwevedi** [comment]
Comment

**Comment by MohitNSamagra** [comment]
- @KDwevedi @VedantKhairnar to update task list

**Comment by karntrehan** [comment]
@sasi2312 to start work on issue event flow with @VedantKhairnar

**Comment by MohitNSamagra** [comment]
- @sasi2312 Dispacher basis X-Github-Event header - 25th Apr

**Comment by VedantKhairnar** [comment]
26th Apr(18:00)
- Go through the handover videos
- Separate out functions into files: issues, installations, comments, PRs
- Put dummy functions in comments and PRs

**Comment by VedantKhairnar** [comment]
1st May(18:26)
Done:
- 5 handlers/transformers created(excluding the parent transformer): installation, issueComments, issues, labels, pull requests 
- DB Adapter using Singleton pattern is done

To do:
- Generic functions for CRUD operations

**Comment by MohitNSamagra** [comment]
- @VedantKhairnar to check with @karntrehan for db changes - to update

**Comment by karntrehan** [comment]
* Issue comment handler completed. 
* DBAdapter completed. 
* Github Adapter completed. 
* Need help from @KDwevedi to figure out dev testing.

**Comment by MohitNSamagra** [comment]
- VC Adapter refactoring - required? @karntrehan to advice

**Comment by karntrehan** [comment]
1. VC to be done separately as a separate ticket. 
2. Unit tests aded. PR raised. To be reviewed by Kanav.
3. Mohit to test post PR merge.

**Comment by MohitNSamagra** [comment]
- @KDwevedi - to review and merge PR

**Comment by MohitNSamagra** [comment]
- Comments to be resolved by @sasi2312  - 17th

**Comment by karntrehan** [comment]
Additional comments raised. To be closed today.

**Comment by MohitNSamagra** [comment]
- 2/4 comments pending - to be closed post #418

**Comment by karntrehan** [comment]
Blocked on https://github.com/orgs/Code4GovTech/projects/1/views/2?pane=issue&itemId=64056153

**Comment by MohitNSamagra** [comment]
- Remaining comments to be resolved during beckn cleanup/deployment
- To be tested partly with DMP and the rest with beckn deployment

**Comment by karntrehan** [comment]
Subsumed under https://github.com/Code4GovTech/server/issues/79
